src/async_buffer_redirect.rs

```diff
@@ -0,0 +1,70 @@
+use gag::Redirect;
+use std::fs::File as StdFile;
+use std::io;
+use std::pin::Pin;
+use std::task::Context;
+use std::task::Poll;
+use tempfile::NamedTempFile;
+use tokio::fs::File;
+use tokio::io::AsyncRead;
+use tokio::io::ReadBuf;
+
+/// Buffer output in an in-memory buffer. Async version of [`gag::BufferRedirect`].
+pub struct AsyncBufferRedirect {
+    #[allow(dead_code)]
+    redir: Redirect<StdFile>,
+    outer: File,
+}
```

Comment:
Exactly the same as `gag`'s `BufferRedirect` type, except it holds a `tokio::fs::File` instead of a `std::fs::File` so I can use `AsyncRead` instead of hacking together some `tokio::task::spawn_blocking` + `Read` thing.